src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -163,20 +165,22 @@ public static boolean addBlobKey(String blobKey, Entity userEntity, DatastoreSer
         return false;
     }
     
-    userEntity.setProperty("blobKey", blobKey);
-    datastore.put(userEntity);
- 
+    User user = new User(userEntity);
+    user.setBlobKey(blobKey);
+
     return true; 
   }
  
   /**
     adds a specified number of points to the user's points
   */
   public static void addPoints(Entity userEntity, int numPoints, DatastoreService datastore){
+    User user = new User(userEntity);
+
     try {
-      userEntity.setProperty("score", ((Number) userEntity.getProperty("score")).intValue() + numPoints);
+      user.setScore(user.getScore() + numPoints);
     } catch (NullPointerException e) {
```

Comment:
Same question as above with the null catches

Reply:
Same issue as above ^ if it's the first time the user is getting points, getScore throws a null pointer exception, but we still want to add points so that's what the catch block is for